mtv-dl: init at 0.23.25

jtrees opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

A command line tool to download videos from public broadcasting services in Germany.

https://github.com/fnep/mtv_dl

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jtrees committed:
python312Packages.durationpy: init at 0.7

Lurkki14 commented on pkgs/development/python-modules/durationpy/default.nix:
```diff
@@ -0,0 +1,17 @@
+    sha256 = "sha256-XvlBa1J7UNci80ZVvs+3Xkkijrgvh7hV7RkRszFLVAg=";
+  };
+
+  meta = with lib; {
+    description = "Module for converting between datetime.timedelta and Go's Duration strings";
+    homepage = "https://github.com/icholy/durationpy";
+    license = licenses.mit;
+  };
```
`meta.maintainers` should be set

jtrees replied:
Yeah. Anyone feel up for it? I really don't know my way around the python ecosystem and don't feel comfortable putting myself in there.

SuperSandro2000 replied:
This is also missing a pythonImportsCheck

jtrees replied:
@Lurkki14 I added myself as a maintainer.

@SuperSandro2000 I added the check.

Lurkki14 commented on pkgs/development/python-modules/durationpy/default.nix:
```diff
@@ -0,0 +1,17 @@
+
+  src = fetchPypi {
+    inherit version pname;
+    sha256 = "sha256-XvlBa1J7UNci80ZVvs+3Xkkijrgvh7hV7RkRszFLVAg=";
```
```suggestion
    hash = "sha256-XvlBa1J7UNci80ZVvs+3Xkkijrgvh7hV7RkRszFLVAg=";
```

jtrees replied:
Updated

Lurkki14 commented on pkgs/tools/misc/mtv-dl/default.nix:
```diff
@@ -0,0 +1,35 @@
+    rich
+  ];
+
+  meta = with lib; {
+    description = "Command line tool to download videos from sources available through MediathekView";
+    homepage = "https://github.com/fnep/mtv_dl";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jtrees ];
+  };
```
`meta.mainProgram` should be set

jtrees replied:
Ah, good point. I added it.

jtrees commented:
@Lurkki14 Thanks for the swift review.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/looking-for-maintainer-for-mtv-dl-python-application/51501/1

SuperSandro2000 commented on pkgs/tools/misc/mtv-dl/default.nix:
```diff
@@ -0,0 +1,36 @@
+python3.pkgs.buildPythonApplication rec {
+  pname = "mtv-dl";
+  version = "0.23.12";
+  format = "pyproject";
```
```suggestion
  pyproject = true;
```

SuperSandro2000 commented on pkgs/tools/misc/mtv-dl/default.nix:
```diff
@@ -0,0 +1,36 @@
+    hash = "sha256-OLDFSoyMztzFs/nMCbLp2WcMJhJ/G3xkkhebe0v8BEo=";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [
+    poetry-core
+    poetry-dynamic-versioning
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
```
```suggestion
  build-system = with python3.pkgs; [
    poetry-core
    poetry-dynamic-versioning
  ];

  dependencies = with python3.pkgs; [
```

jtrees replied:
Updated

SuperSandro2000 commented on pkgs/tools/misc/mtv-dl/default.nix:
```diff
@@ -0,0 +1,36 @@
+    pyyaml
+    rich
+  ];
+
```
Missing pythonImportsCheck

jtrees replied:
I added it.

jtrees commented:
@SuperSandro2000 Thanks you for your review. I integrated the requested changes and would appreciate a re-review.

I also had to update the package version and some of its dependencies to get deps to resolve.

FYI: I played around with getting some unit tests to run, but they require ruff as a python-module which, AFAIK, has not been packaged (only the rust binary part has) in nixpkgs yet and adding that felt let too big of a rabbit hole in this context.

jtrees committed:
mtv-dl: init at 0.24.0

jtrees commented:
I updated the PR with the following changes:

- Rebased on `master`
- Removed obsolete commits for `rich` and `certifi`. They are up-to-date enough now.
- Bumped version of mtv-dl to 0.24.0
- Moved the package into the `by-name` directory

Can we get this merged now?

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5314

tomodachi94 commented:
Looks like there's another merge conflict, happy to take a look after that.

SuperSandro2000 commented:
I think we should rather close this PR and open a new one that follows the current guidelines as otherwise I could probably write a review comment onto every other line of code 😅
